pkg/backingstore/backingstore.go

```diff
@@ -343,21 +343,33 @@ func createCommon(cmd *cobra.Command, args []string, storeType nbv1.StoreType, p
 
 // RunCreateAWSS3 runs a CLI command
 func RunCreateAWSS3(cmd *cobra.Command, args []string) {
+	log := util.Logger()
 	createCommon(cmd, args, nbv1.StoreTypeAWSS3, func(backStore *nbv1.BackingStore, secret *corev1.Secret) {
 		targetBucket := util.GetFlagStringOrPrompt(cmd, "target-bucket")
 		region, _ := cmd.Flags().GetString("region")
 		secretName, _ := cmd.Flags().GetString("secret-name")
+		mandatoryProperties := []string{"AWS_ACCESS_KEY_ID", "AWS_SECRET_ACCESS_KEY"}
 
 		if secretName == "" {
 			accessKey := util.GetFlagStringOrPromptPassword(cmd, "access-key")
 			secretKey := util.GetFlagStringOrPromptPassword(cmd, "secret-key")
 			secret.StringData["AWS_ACCESS_KEY_ID"] = accessKey
 			secret.StringData["AWS_SECRET_ACCESS_KEY"] = secretKey
-		} else {
-			mandatoryProperties := []string{"AWS_ACCESS_KEY_ID", "AWS_SECRET_ACCESS_KEY"}
+		} else {	
 			util.VerifyCredsInSecret(secretName, options.Namespace, mandatoryProperties)
 			secret.Name = secretName
+			secret = util.GetSecretFromSecretReference(&corev1.SecretReference{
+				Name:      secret.Name,
+				Namespace: secret.Namespace,
+			})
+			util.SecretResetStringDataFromData(secret)
+		}
+
+		suggestedSecret := util.CheckForIdenticalSecretsCreds(secret, mandatoryProperties)
```

Comment:
why do you prevent that? and how similarly it's being handled in the reconcilers?
Also, I think that a better behaviour from NooBaa's side is not to fail the command, but just to show the already existing secret name and ask the user if noobaa should proceed creating the backingstore with the already existing secret you found / fail the command.